src/net.h

```diff
@@ -466,7 +466,7 @@ class CNode
      * from the wire. This cleaned string can safely be logged or displayed.
      */
     std::string cleanSubVer GUARDED_BY(m_subver_mutex){};
-    bool m_prefer_evict{false}; // This peer is preferred for eviction.
+    bool m_prefer_evict{false}; // This peer is preferred for eviction. (treated as const)
```

Comment:
As your commit log already says, I think it'd be better just to actually make this const and included as a ctor parameter (and also made private).

Reply:
I guess I'm okay with that in a follow up, but this patchset's been stewing for about 15 months now, so I'm a bit skeptical about increasing the scope?

Reply:
Same as [m_permissionFlags](https://github.com/bitcoin/bitcoin/pull/25174#discussion_r880323950), better make it `const`. It is even easier than `m_permissionFlags` because there are no fuzzer changes. Or should we fuzz `m_prefer_evict` too? It would be one-line extension to the fuzzer's `ConsumeNode()` to pass one more boolean parameter from `ConsumeBool()`.

Reply:
Agree to make these private data members (consider to prefer [non-const](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c12-dont-make-data-members-const-or-references)), if they do not need to be public or protected for testing/fuzzing.

Reply:
Making data members private does not make them thread safe. This member is thread safe in the current code, despite the lack of a guarding mutex, because its value is fixed immediately after creation, before another thread can access it, and it does not change until it's destructed.

Reply:
Yes, the point is to encapsulate if the data member participates in the object’s invariant, in which case it should probably either be const (only if thread-safe) or private.

Reply:
The reason it's thread-safe is that it's (effectively) const; the reason it's (effectively) const is that its value is known at the time the connection is setup. This PR is only about documenting why data members are thread safe, not changing logic or encapsulation...

I don't really get the `private` suggestion -- it's a `CNode` data member, but it's not accessed by any `CNode` member functions, it's only used by `CConnman::AttemptToEvictConnection`. While `CConnman` is a friend class so could get away with that, it seems like a weird thing to do.

https://github.com/ajtowns/bitcoin/commits/202208-cnodeoptions has a commit with these actually made `const` for whatever that's worth. Again, it does more than just document the current state of the code, so I'm not adding it to this PR.

Reply:
Made const in #25962